### PR TITLE
BUG: Fix for #37454: allow reversed axis when plotting with TimedeltaIndex

### LINKED ISSUE #37454: ValueError raised when plottting reversed fixed-frequency TimedeltaIndex
When plotting with a `TimedeltaIndex` with a fixed frequency and `xmin > xmax`, a ValueError gets raised.
```python
>>> pd.Series([1,2], index=pd.timedelta_range(start=0, periods=2, freq='D')).plot(xlim=(3, 1))
```
> C:\ProgramData\Miniconda3\envs\env_name\lib\site-packages\pandas\plotting\_matplotlib\converter.py in __call__(self, x, pos)
>   1066     def __call__(self, x, pos=0) -> str:
>   1067         (vmin, vmax) = tuple(self.axis.get_view_interval())
>-> 1068         n_decimals = int(np.ceil(np.log10(100 * 1e9 / (vmax - vmin))))
>   1069         if n_decimals > 9:
>   1070             n_decimals = 9
>
> ValueError: cannot convert float NaN to integer

Seems that there should just be an `abs` in [this line](https://github.com/pandas-dev/pandas/blob/master/pandas/plotting/_matplotlib/converter.py#L1075).

I would expect this to be able to output a plot successfully.

Works fine with a `TimedeltaIndex` with `freq=None`:
```python
pd.Series([1,2], index=[pd.Timedelta(days=1), pd.Timedelta(days=2)]).plot(xlim=(3, 1))
```

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252

pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.3
setuptools       : 49.6.0.post20201009
Cython           : None
pytest           : 6.1.1
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : 7.18.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 1.0.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.0
sqlalchemy       : 1.3.20
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

### PR DESCRIPTION
**Pull request by theavey**
- [x] closes #37454
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.4.rst [hunk 1]
```diff
@@ -41,6 +41,7 @@ Bug fixes
 - Bug in :meth:`GroupBy.fillna` that introduced a performance regression after 1.0.5 (:issue:`36757`)
 - Bug in :meth:`DataFrame.info` was raising a ``KeyError`` when the DataFrame has integer column names (:issue:`37245`)
 - Bug in :meth:`DataFrameGroupby.apply` would drop a :class:`CategoricalIndex` when grouped on (:issue:`35792`)
+- Bug in :meth:`Series.plot` that would break if ``xmin > xmax`` when using a fixed frequency :class:`TimedeltaIndex` (:issue:`37454`)
```
Comment (arw2019) [inline doc/source/whatsnew/v1.1.4.rst:44]:
```
Bug in :meth:`Series.plot` was throwing ``ValueError`` with a :class:`Series` indexed by a :class:`TimedeltaIndex` with a fixed frequency when x-axis lower limit was greater than upper limit  (:issue:`37454`)
```

**Review by arw2019 — CHANGES_REQUESTED** [review]
Thanks @theavey for the PR!

A suggestion re: wording on the whatsnew note, otherwise LGTM

File: doc/source/whatsnew/v1.1.4.rst (hunk 1 above)
Comment (theavey) [reply doc/source/whatsnew/v1.1.4.rst:44]: yeah, definitely a bit clearer. Wasn't sure how much I should value brevity, but I'll make that change

**Review by arw2019 — APPROVED** [review]

File: doc/source/whatsnew/v1.1.4.rst [hunk 2]
```diff
@@ -41,6 +41,8 @@ Bug fixes
 - Bug in :meth:`GroupBy.fillna` that introduced a performance regression after 1.0.5 (:issue:`36757`)
 - Bug in :meth:`DataFrame.info` was raising a ``KeyError`` when the DataFrame has integer column names (:issue:`37245`)
 - Bug in :meth:`DataFrameGroupby.apply` would drop a :class:`CategoricalIndex` when grouped on (:issue:`35792`)
+- Bug in :meth:`Series.plot` and :meth:`DataFrame.plot` was throwing :exc:`ValueError` with a :class:`Series` or :class:`DataFrame`
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.4.rst:44]: move to 1.2 under the visualization section.

**Review by jreback — CHANGES_REQUESTED** [review]
can you add the actual plotting tests from the OP (both working and non-working)
pandas/tests/plotting/tests_series.py is prob fine.
pls merge master and ping on green.

File: pandas/tests/plotting/test_converter.py [hunk 3]
```diff
@@ -361,3 +361,13 @@ def test_format_timedelta_ticks(self, x, decimal, format_expected):
         tdc = converter.TimeSeries_TimedeltaFormatter
         result = tdc.format_timedelta_ticks(x, pos=None, n_decimals=decimal)
         assert result == format_expected
+
+    @pytest.mark.parametrize("view_interval", [(1, 2), (2, 1)])
+    def test_call(self, view_interval, monkeypatch):
```
Comment (jreback) [inline pandas/tests/plotting/test_converter.py:366]: can you add the issue number as a comment; also can you give the test a meaningful name

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm with comment. ping on greenish.

File: pandas/tests/plotting/test_converter.py (hunk 3 above)
Comment (theavey) [reply pandas/tests/plotting/test_converter.py:366]: yeah, will do.

File: doc/source/whatsnew/v1.1.4.rst (hunk 2 above)
Comment (theavey) [reply doc/source/whatsnew/v1.1.4.rst]: moved to v1.2.0 in the plotting section

**Comment by theavey** [comment]
@jreback, looks like everything is working except some unrelated failures with FutureWarning.

**Comment by theavey** [comment]
@jreback, I think this is good to go

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @theavey very nice

**Pull request merged**